docs/server/source/http-client-server-api.rst

```diff
@@ -671,7 +674,7 @@ Blocks
 
    :resheader Content-Type: ``application/json``
 
-   :statuscode 200: A list of blocks containing a transaction with ID ``transaction_id`` was found and returned.
+   :statuscode 200: A list containing atmost one block height where ``transaction_id`` was found and returned.
```

Comment:
atmost --> at most

Reply:
I wonder what happens if two or more block heights are found? Would it just happily return the list and a 200 status code.

It seems to me that something is seriously wrong if a list of more than one block height comes back. Maybe a CRITICAL logging message should be written to the logs, and the HTTP response could ask them to report the incident to the network operators?

Reply:
Under normal operation that should not happen but if it does its an indication of some malicious activity on the node or something is critically wrong. Either way it would be good to generate a log message.

Reply:
I see that you added the CRITICAL log message if the response list has more than one block ID.

I guess what will happen is that the HTTP response will contain a list with more than one block ID, and the response status code will be 200. Maybe the description of that 200 status code could be changed a bit, to reflect that possibility.

The status code 200 can also happen if there are no blocks containing a transaction with that transaction ID.

Maybe

```text
   :statuscode 200: The request was properly formed and zero or more blocks were found containing the specified ``transaction_id``.
```

Maybe the sentence describing this endpoint ("Retrieve a list of block IDS...") could be followed by a second sentence noting that the list should be either empty or contain only one block ID.